### PR TITLE
refactor interceptors logger and secret getter

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

This is a refactoring PR that does two things:
1. It removes the unused logger from the core interceptors
2. It adds a DefaultSecretGetter with default values for the cache size and ttl (we were using the same cache args in a bunch of places)

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/interceptors/secret_getter.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package interceptors
```
Comment (savitaashture) [inline pkg/interceptors/secret_getter.go:1]: Could you add a license

File: pkg/interceptors/secret_getter_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package interceptors_test
```
Comment (savitaashture) [inline pkg/interceptors/secret_getter_test.go:1]: here as well

**Review by savitaashture — APPROVED** [review]

**Review by khrm — COMMENTED** [review]
/approve

**Review by khrm — COMMENTED** [review]
/lgtm

**Pull request merged**